tools/util_windows.go

```diff
@@ -0,0 +1,163 @@
+// +build windows
+
+package tools
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"unsafe"
+
+	"github.com/rubyist/tracerx"
+	"golang.org/x/sys/windows"
+)
+
+var (
+	availableClusterSize = []int64{64 * 1024, 4 * 1024} // ReFS only supports 64KiB and 4KiB cluster.
+	GiB                  = int64(1024 * 1024 * 1024)
+)
+
+// _FSCTL_DUPLICATE_EXTENTS_TO_FILE :
+// Instructs the file system to copy a range of file bytes on behalf of an application.
+//
+// https://docs.microsoft.com/windows/win32/api/winioctl/ni-winioctl-fsctl_duplicate_extents_to_file
+const _FSCTL_DUPLICATE_EXTENTS_TO_FILE = 623428
+
+// _DUPLICATE_EXTENTS_DATA :
+// Contains parameters for the FSCTL_DUPLICATE_EXTENTS control code that performs the Block Cloning operation.
+//
+// https://docs.microsoft.com/windows/win32/api/winioctl/ns-winioctl-duplicate_extents_data
+type _DUPLICATE_EXTENTS_DATA struct {
+	FileHandle       windows.Handle
+	SourceFileOffset int64
+	TargetFileOffset int64
+	ByteCount        int64
+}
+
+// CheckCloneFileSupported runs explicit test of clone file on supplied directory.
+// This function creates some (src and dst) file in the directory and remove after test finished.
+//
+// If check failed (e.g. directory is read-only), returns err.
+func CheckCloneFileSupported(dir string) (supported bool, err error) {
+	src, err := ioutil.TempFile(dir, "src")
+	if err != nil {
+		return false, err
+	}
+	defer os.Remove(src.Name())
+
+	_, err = src.WriteString("TESTING") // Non-empty file is required for correct result.
```

Comment:
Is a non-empty file required just for this case or for the function itself to work correctly?  If the latter, we probably want to check for a non-zero size below.

Reply:
Calling `FSCTL_DUPLICATE_EXTENTS_TO_FILE` to empty file is always success even filesystem don't support it.
To check filesystem support `FSCTL_DUPLICATE_EXTENTS_TO_FILE` or not, we have to use non-empty file.

I'll write those as comment.